src/S3WriterException.php

```diff
@@ -0,0 +1,27 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\S3Writer;
+
+use Aws\S3\Exception\S3Exception;
+use GuzzleHttp\Exception\ClientException;
+use Keboola\Component\UserException;
+
+class S3WriterException
+{
+    public static function factory(S3Exception $e): UserException
+    {
+        /** @var ClientException $previous */
+        $previous = $e->getPrevious();
+        return new UserException(
```

Comment:
Akorát tady je to IMO špatně - nemůžeš si být jistý, že je to ClientException, ne? Měl by tam být runtime check IMO.